### ISSUE TITLE
Slug has already been taken error when using globalize

### ISSUE THREAD

**Issue by mgharbik** [solidusio/solidus#1243]
Hi,

Cloning a product raises an error slug has already been taken. `friendly_id` normally appends a :uuid code to the slug if it was used before, probably with globalize it gives this issue?

![slug](http://img15.hostingpics.net/pics/605397Capturedecran20160614a150059.png)

Note: Last day my colleague suggested that it will be nice to generalize the slug, that all contain the :uuid code, even the new ones. So that products would have the same preferability by search engines. Hope that helps.

With kind regards,

**Comment by jasonfb** [comment]
would you explain what you mean by "generalize the slug" -- do you mean denormalize (move data into a separate table) ? Sorry I think some developers use terms differently that's why I'm asking-- just to make sure I understand your meaning.

**Comment by simo163** [comment]
@jasonfb Sorry for the confusion! normally when we clone a product we got the following slugs:

- original product: apache-baseball-jersey
- cloned product: apache-baseball-jersey-26fff430-9ee0-4dad-909a-af15329d77ad

I meant by "generalize the slug" (maybe the verb or expression is not appropriate), just generate the :uuid code in both, so the slugs would be for example:

- original product: apache-baseball-jersey-a877c085-1943-41f6-a778-75240f94bcce
- cloned product: apache-baseball-jersey-26fff430-9ee0-4dad-909a-af15329d77ad

With kind regards,

**Comment by mamhoff** [comment]
I would like a shop without cloned products to still be able to keep a non-slugged URL for a product page. Is this an issue only when used with `globalize`?

**Comment by jrochkind** [comment]
I would not want all slugs by default to have ugly UUIDs in them, that's not a very "friendly" id at all.

**Comment by tvdeyen** [comment]
I think it's very unlikely, that cloning a normal product getting the same slug as another product. This can't happen. I guess there is some issue with `solidus-globalize`

We don't want uuids in every product slug. 

This sounds like a `friendly_id` issue, not a Solidus one. Can you confirm this only happens with `solidus-globalize` activated? Then the issue should be opened on that repo.

Thanks for reporting.

**Comment by simo163** [comment]
I confirm I have this issue with ```solidus_globalize```activated. I also believe there is an issue with the gem, I tried to install it in a new project but it gives:

```
/Users/my-machine/.rvm/gems/ruby-2.2.1/bundler/gems/solidus_globalize-02ed2c1d3937/app/models/spree/store_decorator.rb:3:in `block in <module:Spree>': undefined method `translates' for Store:Module (NoMethodError)
```

without ```solidus_globalize``` the error is raised only if the product is cloned twice.

**Comment by jasonfb** [comment]
I think my marketing team would also concur with 👎  on inelegant UUIDs in Urls. It seems to me that a normal workflow would involved copying the product, changing its slug to one that is in fact unique, and then copying it again, thus eliminating the double-copy uniqueness problem.

**Comment by dangerdogz** [comment]
I think this might be related to the following bug :

FriendlyID is using the **current locale** language to validate uniqueness of the slug, so if you're navigating in french, you can edit the english slugs and FriendlyID is gonna compare the proposed value to all the french slugs so you can set duplicate values.

Example:

Navigating in english, I have a Ruby on Rails bag with a ruby-on-rails-bag slug

<img width="1018" alt="screen shot 2016-06-28 at 8 38 26 pm" src="https://cloud.githubusercontent.com/assets/4110905/16436940/54e142c4-3d70-11e6-97e2-f544b3766996.png">

Then I open the Ruby on Rails mug with a ruby-on-rails-mug slug

<img width="1016" alt="screen shot 2016-06-28 at 8 39 21 pm" src="https://cloud.githubusercontent.com/assets/4110905/16436955/76acdd50-3d70-11e6-99e4-3783e16e7134.png">

I replace the english slug (ruby-on-rails-mug) while my store is in english with the ruby-on-rails-bag slug (which already exists) and I try to update, I will get a slug already taken error

<img width="1068" alt="screen shot 2016-06-28 at 8 56 21 pm" src="https://cloud.githubusercontent.com/assets/4110905/16437220/d02c9ba2-3d72-11e6-92d4-7e4d10ae4c50.png">

If I retry the same thing while my Solidus is in french, I can set the english slug of ruby-on-rails-mug to ruby-on-rails-bag because FriendlyId is comparing the english value I'm trying to set with the french slugs of the other products

<img width="1134" alt="screen shot 2016-06-28 at 8 57 35 pm" src="https://cloud.githubusercontent.com/assets/4110905/16437246/12c00a9e-3d73-11e6-985e-16d8f0cdb336.png">

And now the english link for the Mug is the same as the one for the bag

<img width="539" alt="screen shot 2016-06-28 at 8 59 48 pm" src="https://cloud.githubusercontent.com/assets/4110905/16437274/540ce134-3d73-11e6-98ad-b6e5ef8eecbd.png">

TL;DR : You can have duplicate slug by updating a product slug in a different language than the one you're using for browsing because FriendlyID is using your browsing locale to validate uniqueness instead of using the language of the updated slug

**Comment by kennyadsl** [comment]
Closing due to inactivity
